### DISCUSSION TITLE
net6.0-macos publish creating .pkg file instead of .app

### DISCUSSION THREAD

**Discussion by natan-abolafya** [dotnet/runtime#75346] [Q&A]
Is this possible? We have been using the output of `dotnet build` for this but I'd like to try trimming to lower the artifact size. But when I change to `dotnet publish` it produces a pkg file instead. Is there a flag that I can use?

I came across this https://docs.microsoft.com/en-us/dotnet/maui/macos/deployment/overview and tried `/p:CreatePackage=false` but that didn't help. It's not a MAUI project either anyway.

**Comment by vitek-karas** [comment]
What type of app are you building? (console, ASP.NET, ...)
What is the publish command?
Is it an "application" (OutputType=exe)?

**Reply by natan-abolafya** [reply]
console, OutputType=exe

`dotnet publish /p:Version="1.0.0-156" /p:Configuration='Release' Client.csproj`. Tried running it on `.sln` file also but that didn't help.

**Reply by vitek-karas** [reply]
That is really weird - can you repro it on another project? (like take a simple Hello World for example, or something like that).
Do you specify `RuntimeIdentifier` somewhere?
Is it self-contained or not?

(Obviously if you'd have a repro project that would be great)

**Reply by natan-abolafya** [reply]
We have this in the csproj file.
```
<RuntimeIdentifiers>osx-arm64;osx-x64</RuntimeIdentifiers>
```

We do not pass anything for the self-contained, neither on csproj nor the publish command. But we have been using the artifacts as self-contained. Which in hindsight, sounds wrong? bit it works :)

I will try a hello-world setup,

**Reply by vitek-karas** [reply]
Specifying RuntimeIdentifier (RID) will by default turn on self-contained (unfortunately). So it is self-contained.
Whether you want it or not depends - there are cons/pros for either.
Trimming only works on self-contained for now, but then if you use framework dependent you need much less.

**Reply by vitek-karas** [reply]
I forgot to mention that `publish` on a normal console app should not produce `.app` either. I would expect it to produce a directory with all the necessary files in it - including the executable.

**Reply by natan-abolafya** [reply]
ah, good to know. thanks.

**Reply by natan-abolafya** [reply]
Sorry, I was mistaken about Console application. Rider says Console application but it is 

```
<TargetFramework>net6.0-macos</TargetFramework>
```

When I tested with `net6.0`, it behaved like you said. But with `net6.0-macos` it creates a package with this name `<project-name>-1.0.1..pkg` under the publish folder. Yes double dots at the end.

Here is a small csproj file where you can create a file with `Main` and try out.

```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net6.0-macos</TargetFramework>
        <ApplicationId>com.example.app</ApplicationId>
        <RuntimeIdentifier>osx-arm64</RuntimeIdentifier>
        <SupportedOSPlatformVersion>10.15</SupportedOSPlatformVersion>
    </PropertyGroup>
</Project>
```

**Reply by vitek-karas** [reply]
I actually don't know why that's happening. Could you please open an issue in the dotnet/sdk repo? That's where this area lives.

**Reply by natan-abolafya** [reply]
Done https://github.com/dotnet/sdk/issues/27841. Thanks for the help.

**Comment by NN---** [comment]
I use [DotnetBundle](https://github.com/egramtel/dotnet-bundle) package to build app.
Didn’t know dotnet tool is capable to perform this.
